modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -118,5 +127,30 @@ def enum_zones
       logger.debug "Enumerated dns zones: #{zones}"
       zones
     end
+
+    def get_ptr_ip(zone, ip)
+      ip6 = ip.include?(':')
```

Comment:
You are assuming that ipv6 ptr zone boundary is on each four bytes, while it's on a nibble (each four bits/a single hex digit). This is complicated by ipv6 that can be passed in in a shortened format. 

I think what needs to be done is:
1.  incoming address converted to an IPAddr
2.  reverse zone converted to an address with mask and converted to an IPAddr 
3.  check that passed in address belongs to the subnet (IPAddr#include?)
4.  calculate address bit mask: ptr zone address mask XOR 0xffffffff for ipv4 (0xffffffffffffffffffffffffffffffff for ipv6)
5.  using address from #1 and bit mask from #4: IPAddr.new(ip).to_i & mask
6.  split address from #5 into octets for ipv4, nibbles for ipv6, reverse the order, and return